ipam/pool.go

```diff
@@ -423,22 +426,36 @@ func (ap *addressPool) newAddressRecord(addr *net.IP) (*addressRecord, error) {
 // Requests a new address from the address pool.
 func (ap *addressPool) requestAddress(address string, options map[string]string) (string, error) {
 	var ar *addressRecord
+	var addr *net.IPNet
+	var err error
+	id := options[OptAddressID]
+
+	log.Printf("[ipam] Requesting address with address:%v options:%+v.", address, options)
+	defer func() { log.Printf("[ipam] Address request completed with address:%v err:%v.", addr, err) }()
 
 	if address != "" {
 		// Return the specific address requested.
 		ar = ap.Addresses[address]
 		if ar == nil {
-			return "", errAddressNotFound
+			err = errAddressNotFound
+			return "", err
 		}
 		if ar.InUse {
-			return "", errAddressInUse
+			err = errAddressInUse
```

Comment:
This will break idempotency. If clients want to use a specific ip, they should use reservation id, and we should check if the new request is against same id, we are good.